### PR TITLE
chore: DBTP-1011 signpost autogenerated files

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
Addresses [DBTP-1011](https://uktrade.atlassian.net/browse/DBTP-1011).

---
### Title:
- [x] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [x] Link to ticket included (unless it's a quick out of ticket thing)
- [x] Includes tests (or an explanation for why it doesn't)
- [ ] If the work includes user interface changes, before and after screenshots included in description
- [ ] Includes any applicable changes to the documentation in this code base
- [ ] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)
### Tasks:
- [ ] [Trigger the pull request regression tests for this branch](https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests) and confirm that they are passing
- [ ] Manually run the commented out sections of the [pull request regression tests](https://github.com/uktrade/platform-tools/blob/main/regression_tests/pull_request_tests.sh) (unless it's not a code change)

[DBTP-1011]: https://uktrade.atlassian.net/browse/DBTP-1011?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by WillGibson** [comment]
This is looking good subject to the line breaks question and also including the following auto generated files...

* `copilot/web/overrides/cfn.patches.yml`
* `copilot/pipelines/application/*` Perhaps could be achieved simply by dropping a text file into the root of that directory if do all the files is tricky

**Comment by samuele-mattiuzzo** [comment]
> This is looking good subject to the line breaks question and also including the following auto generated files...
> 
> * `copilot/web/overrides/cfn.patches.yml`
> * `copilot/pipelines/application/*` Perhaps could be achieved simply by dropping a text file into the root of that directory if do all the files is tricky

The other option would be to modify our `.yml` template files directly so that `{% version_tag %}` only prints out the version, and add the warning line manually in them.

Which, in hindsight, is probably the most tedious yet the best way forward

**Comment by WillGibson** [comment]
> The other option would be to modify our `.yml` template files directly so that `{% version_tag %}` only prints out the version, and add the warning line manually in them.
> 
> Which, in hindsight, is probably the most tedious yet the best way forward

Wouldn't this suggestion result on a load of duplicated boiler plate?

**Comment by samuele-mattiuzzo** [comment]
> > The other option would be to modify our `.yml` template files directly so that `{% version_tag %}` only prints out the version, and add the warning line manually in them.
> > Which, in hindsight, is probably the most tedious yet the best way forward
> 
> Wouldn't this suggestion result on a load of duplicated boiler plate?

@WillGibson can you give it another pair of eyes? I've added a new template tag that does the multiline trick.

@gabelton yes, maybe, however it's not straight forward to have a multiline template tag that is yaml-compliant, so for now I'm perfectly happy if this stays as is as I don't want it to take days

File: dbt_platform_helper/templates/svc/manifest-public.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# {% extra_header %}
```
Comment (gabelton) [inline dbt_platform_helper/templates/svc/manifest-public.yml:1]: Do we want this in this file and manifest-backend? We do manually edit those don't we? i.e. adding ip-filter or tweaking env vars etc.

**Comment by gabelton** [comment]
> @gabelton yes, maybe, however it's not straight forward to have a multiline template tag that is yaml-compliant, so for now I'm perfectly happy if this stays as is as I don't want it to take days

Could you add something on the same line or does that break linting rules? Otherwise yeah not a biggy

**Comment by samuele-mattiuzzo** [comment]
> > @gabelton yes, maybe, however it's not straight forward to have a multiline template tag that is yaml-compliant, so for now I'm perfectly happy if this stays as is as I don't want it to take days
> 
> Could you add something on the same line or does that break linting rules? Otherwise yeah not a biggy

I could yes, it's not a biggie at all as it's just a comment, it's just ugly to have a too long line. Ideally, the new template tag I added would be able to spit out content with line breaks, but that's a future enhancement I suppose, to keep this ticket's scope contained

File: dbt_platform_helper/templates/svc/manifest-public.yml (hunk 1 above)
Comment (samuele-mattiuzzo) [reply dbt_platform_helper/templates/svc/manifest-public.yml:1]: Yes, true, I think I went a bit too far with the find / replace. Will update

**Review by WillGibson — APPROVED** [review]
Beautifully simple solution in the end :-)

**Review by gabelton — APPROVED** [review]

**Pull request merged**